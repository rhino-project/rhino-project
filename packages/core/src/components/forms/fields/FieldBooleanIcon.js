import PropTypes from 'prop-types';
import { useController } from 'react-hook-form';
import { useFieldInheritedProps } from '../../../hooks/form';
import { Icon } from '../../icons';
import { useGlobalComponent } from '../../../hooks/overrides';

export const FieldBooleanIconBase = ({
  trueIcon = 'check',
  falseIcon = 'x',
  ...props
}) => {
  const { extractedProps, inheritedProps } = useFieldInheritedProps(props);
  const { path } = props;
  const {
    field: { value, onChange, ...fieldProps }
  } = useController({
    name: path
  });

  return (
    <Icon
      {...extractedProps}
      icon={value ? trueIcon : falseIcon}
      onClick={() => onChange(!value)}
      {...fieldProps}
      {...inheritedProps}
    />
  );
};

FieldBooleanIconBase.propTypes = {
  path: PropTypes.string.isRequired
};

export const FieldBooleanIcon = (props) =>
  useGlobalComponent('FieldBooleanIcon', FieldBooleanIconBase, props);
