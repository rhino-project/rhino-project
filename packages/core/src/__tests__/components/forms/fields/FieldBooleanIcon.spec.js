import FieldBooleanIcon from 'rhino/components/forms/fields/FieldBooleanIcon';
import { sharedFieldTests } from './sharedFieldTests';

describe('FieldBooleanIcon', () => {
  sharedFieldTests(FieldBooleanIcon);
});
