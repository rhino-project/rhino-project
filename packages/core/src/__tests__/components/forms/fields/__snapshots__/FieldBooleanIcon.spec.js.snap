// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FieldBooleanIcon > render with place holder 1`] = `
<DocumentFragment>
  <svg
    class=""
    fill="currentColor"
    height="32"
    id="dummy"
    name="dummy"
    path="dummy"
    placeholder="placeholder"
    width="32"
  >
    <use
      xlink:href="/src/assets/images/bootstrap-icons.svg#x"
    />
  </svg>
</DocumentFragment>
`;

exports[`FieldBooleanIcon > renders inside of model context 1`] = `
<DocumentFragment>
  <svg
    class=""
    fill="currentColor"
    height="32"
    id="dummy"
    name="dummy"
    path="dummy"
    width="32"
  >
    <use
      xlink:href="/src/assets/images/bootstrap-icons.svg#x"
    />
  </svg>
</DocumentFragment>
`;

exports[`FieldBooleanIcon > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
